server/utils/socketAction.js

```diff
@@ -0,0 +1,16 @@
+const { emit } = require('../utils');
+
+const waitForAction = (socket, event, requestAction, receiveAction) => (
```

Comment:
The websocket emits _events_ on a _channel_, so I'd say the function should be named `waitForEvent` and have `channel` as one of its parameters (instead of event as it has now).

Considering that we use "action" as the name for everything that goes through redux, I'm okay with leaving it as is.

This would also do something with the weird naming like `socket.createAction.action`, which probably should be `socket.createEvent.action` (or `auth` or `admin`), if I'm not mistaken.

TL;DR: naming things is hard

Reply:
In this case it is literally waiting for an action though. Yes, it uses events to do that, but it only cares about a single action from that event listener. 
And regarding `channel`: both Nodejs and socketio use `eventName` to refer to what I have called `event`. I will consider changing it to eventName though.

I agree about socket.createAction. That started as an ugly hack that grew uglier. I think I'll rename it to emitEvent. Preferably it would just be named emit, but we have to mock that function for testing.